### ISSUE TITLE
borg serve --append-only

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#1168]
Extracting the easy part from #1164:

We have this only as repo config option, which is a bit awkward to control (you need to edit the config).

If `borg serve` had this option, it could be used in `.ssh/authorized_keys` like this:
```
command="borg serve --append-only ..." ssh-rsa <key used for not-always-trustable backup clients>
command="borg serve ..." ssh-rsa <key used for backup management>
```

***

:moneybag: [there is a bounty for this](https://www.bountysource.com/issues/35160230-borg-serve-append-only)

**Comment by PlasmaPower** [comment]
Should I extend the repository options [here](https://github.com/borgbackup/borg/blob/master/src/borg/remote.py#L130) or will that break things? Maybe restrict the RPC endpoints?

**Comment by ThomasWaldmann** [comment]
I had a quick look and yes, adding another parameter to `Repository.__init__` seems ok.

You can also have a look at how `restrict_to_paths` is set up / used and add support for append_only in a similar way.

**Comment by ThomasWaldmann** [comment]
What do you think, shall we move this to 1.0-maint due to the security implications ("practically usable security")?

**Comment by ThomasWaldmann** [comment]
@enkore what do you think? see my last comment.

**Comment by enkore** [comment]
It certainly makes things easier to use and the patch is quite small. So, yeah, why not, we got a patch, it won't delay stuff.

**Comment by PlasmaPower** [comment]
Hmm, are you sure "fixes" works? The issue isn't closed. I've always used "resolves".

**Comment by enkore** [comment]
Github will close the issue when it's merged into master

**Comment by PlasmaPower** [comment]
Oh, okay. Makes sense.

**Comment by ThomasWaldmann** [comment]
@PlasmaPower please PR again, without .egg stuff.

**Comment by ThomasWaldmann** [comment]
Fixed in 1.0-maint, will soon also get merged into master.
